model/jpa/src/main/java/org/keycloak/models/jpa/entities/GroupEntity.java

```diff
@@ -36,7 +36,8 @@
         @NamedQuery(name="getGroupCountByNameContainingFromIdList", query="select count(u) from GroupEntity u where u.realm = :realm and lower(u.name) like lower(concat('%',:search,'%')) and u.id in :ids"),
         @NamedQuery(name="getTopLevelGroupIds", query="select u.id from GroupEntity u where u.parentId = :parent and u.realm = :realm order by u.name ASC"),
         @NamedQuery(name="getGroupCount", query="select count(u) from GroupEntity u where u.realm = :realm"),
-        @NamedQuery(name="getTopLevelGroupCount", query="select count(u) from GroupEntity u where u.realm = :realm and u.parentId = :parent")
+        @NamedQuery(name="getTopLevelGroupCount", query="select count(u) from GroupEntity u where u.realm = :realm and u.parentId = :parent"),
+        @NamedQuery(name="getGroupIdByName", query="select u.id from GroupEntity u where u.realm = :realm and u.parentId = :parent and u.name = :name")
```

Comment:
I'm fine with rename `getGroupByName` to `getGroupByNameAndParent`.  I guess the lookup of the id is needed because the entity must fetched from the cache if there is an entry in the cache instead of db. I'm doing the lookup like the realmById and realmByName

Reply:
Yes, true. Good point with the lookup by id. Thanks for checking the other places in the codebase, which is something, which I should do as well before making this comment :-)

Reply:
+1 for rename to `getGroupByNameAndParent`